File: ShipManagement/ShipApp/src/app/Models/SalarySummary.ts

export class SalarySummary {

    salarySummaryId?: number | undefined;
    crewId?: number | undefined;
    receivedDate?: Date | undefined;
    createdDate?: Date | undefined;
    updatedDate?: Date | undefined;
    amountTaken?: number|undefined;
    description :string|undefined;
    SalaryId :number|undefined;
    /**
     *
     */
    constructor();
    constructor(Id?:number,year?:number,sdate?:Date,uDate?:Date,rDate?:Date,Cid?:number,sal?:number,bid?:number,desc?:string,sid?:number) {
        this.salarySummaryId =Id ?? 0;
        this.receivedDate =rDate ?? undefined;
        this.createdDate = sdate ?? undefined;
        this.updatedDate =uDate ?? undefined;
        this.crewId =Cid ?? 0;
        this.amountTaken =sal ??0;
        this.description =desc;
        this.SalaryId= sid ??0;
    }
}

File: ShipmanagementAPI/DataLayer/Entities/SalarySummary.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace DataLayer.Entities
{
    public class SalarySummary
    {
        public int SalarySummaryId { get; set; }
        public required decimal AmountTaken { get; set; }
        public required string Description { get; set; }
        public required DateTime ReceivedDate { get; set; }
        public DateTime CreatedDate { get; set; }
        public required DateTime UpdatedDate { get; set; }
        // Navigation properties

        #region Crew to SalarySummary(one-to-Many)
        /* Crew Member to SalarySummary(one-to-Many)
         * A Single Crew can withdraw multiple Payments (Paid record).
         * Each paid record is associated with single Crew Member
         * This is classic one-to-many relationship.
         */
        // Crew member can have multiple leave summary and salary summary
        //Foreign Key
        [ForeignKey("CrewId")]
        public int CrewId { get; set; }
        public CrewInformation? Crew { get; set; }  // This will hold the related Crew entity

        [ForeignKey("Salaries ")]
        public int SalaryId { get; set; }
        public SalaryInformation? Salaries { get; set; }
        #endregion

        public SalarySummary()
        {

        }

    }
}
